src/dataStore/DripsDataStore.sol

```diff
@@ -0,0 +1,240 @@
+            //---------------------------------------------------------------------------------//
+            // Opcode  | Opcode + Arguments  | Description  | Stack View                       //
+            //---------------------------------------------------------------------------------//
+            // 0x60    |  0x600B             | PUSH1 11     | dataOffset                       //
+            // 0x59    |  0x59               | MSIZE        | 0 dataOffset                     //
+            // 0x81    |  0x81               | DUP2         | dataOffset 0 dataOffset          //
+            // 0x38    |  0x38               | CODESIZE     | codeSize dataOffset 0 dataOffset //
+            // 0x03    |  0x03               | SUB          | dataSize 0 dataOffset            //
+            // 0x80    |  0x80               | DUP          | dataSize dataSize 0 dataOffset   //
+            // 0x92    |  0x92               | SWAP3        | dataOffset dataSize 0 dataSize   //
+            // 0x59    |  0x59               | MSIZE        | 0 dataOffset dataSize 0 dataSize //
+            // 0x39    |  0x39               | CODECOPY     | 0 dataSize                       //
+            // 0xF3    |  0xF3               | RETURN       |                                  //
+            //---------------------------------------------------------------------------------//
+            hex"600B5981380380925939F3",
+            hex"00", // Prefix the data with the STOP opcode to ensure that it can't be called.
+            data
+        );
+        // slither-disable-next-line assembly
+        assembly ("memory-safe") {
+            pointer := create(0, add(creationCode, 32), mload(creationCode))
+        }
+        require(pointer != address(0), "Storing data failed");
+    }
+
+    /// @notice Loads the stored data and returns it.
+    /// This function never returns, it stops the execution in the current context,
+    /// and the loaded data is returned as the external function's return data.
+    /// There is no validation, so the loaded data must be ABI-encoded and must
+    /// match exactly the returned data type of an external function that calls this function.
+    /// E.g. if a pointer stores ABI-encoded `uint[]`, ONLY an external function
+    /// returning `uint[]` can call `_loadAndReturn` for that pointer.
+    /// This is cheaper than loading data, ABI-decoding it, and then re-ABI-encoding to return it.
+    /// @param pointer The contract to load data from, see `_storeCalldata`. Reverts if `0`.
+    function _loadAndReturn(address pointer) internal view {
+        require(pointer != address(0), "Requested data not in storage");
+        bytes memory data = pointer.code;
+        // slither-disable-next-line assembly
+        assembly ("memory-safe") {
+            // Skip the first byte, it's the STOP opcode prefix.
+            return(add(data, 33), sub(mload(data), 1))
```

Comment:
```suggestion
            return(add(data, 33), sub(mload(0x40), 1))
```
You could also use the free memory pointer here `0x40`.  would work as well.

Maybe a bit more comments? Explaining that you skip the first slot of 32 bytes.

Reply:
Okay, so the `return` opcode has two params 
- starting offset
- length

Why does `data` work here? In my understanding the code is now in memory and `data` is a pointer to the start of the code.

Reply:
Ahh got it. The first positions of data in memory is the array length.

Reply:
This part could be implemented in many ways, depending on how much assembly we want. I wanted to reduce and encapsulate it as much as possible, so it's a single step, carving out the relevant data from memory and returning it. The "optimal" approach would be probably to do `codecopy` in YUL skipping the first byte, dump it to memory wherever, even on `0x00`, and pass that to `return`. It wouldn't save much gas though, but would be even more confusing than what we have here.

Reply:
This makes sense.  I think the current implementation is a good one.